lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -229,6 +229,61 @@ static Value createDecoder(Value input, unsigned width, Location insertLoc,
   return rewriter.create<DShlPrimOp>(insertLoc, resultType, bit, input);
 }
 
+/// Return the number of bits needed to index the given number of values.
+static size_t getNumIndexBits(ArrayRef<Value> values) {
+  return static_cast<size_t>(
```

Comment:
I'd recommend using Log2_32_Ceil or something similar in llvm/Support/MathExtras.h, it is a lot more efficient and simpler.